### PR TITLE
HIVE-24348: Changes for running beeline with java (Naveen Gangam)

### PR DESCRIPTION
**Pull request by nrg4878**
1) Changes to use java to execute beeline if HADOOP_HOME is not set
            2) POM changes to build an uber beeline-with-dependencies jar
            3) Eliminates some of the not required jars in classpath for beeline.

### What changes were proposed in this pull request?
            1) Changes to use java to execute beeline if HADOOP_HOME is not set
            2) POM changes to build an uber beeline-with-dependencies jar
            3) Eliminates some of the not required jars in classpath for beeline.

### Why are the changes needed?
Currently beeline has a bunch of dependencies on other jars, some required and some unnecessary. beeline script also uses HADOOP_HOME to execute beeline class. This adds some other jars to the classpath that makes it hard to run beeline on an edge node. This fix is a first pass at building a standalone jar with all needed dependencies and also make it work using "java" when HADOOP_HOME is not set.

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
Manually.

### CODE REVIEW

**Comment by belugabehr** [comment]
@nrg4878 Can you also take a look at my PR #1833 since you seem to be looking at beeline improvements?

File: beeline/pom.xml [hunk 1]
```diff
@@ -197,6 +197,25 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-assembly-plugin</artifactId>
+        <version>3.1.1</version>
```
Comment (belugabehr) [inline beeline/pom.xml:203]: I don't think you need to include an explicit version number here.

**Review by belugabehr — CHANGES_REQUESTED** [review]

File: beeline/pom.xml (hunk 1 above)
Comment (nrg4878) [reply beeline/pom.xml:203]: Thanks David. These changes are not final. I am having a new team member look at these changes. There is more work to be done in terms of building a beeline standalone distributable (.zip file). I am just transitioning these code changes for him to take a look.

But your comment is noted.

**Comment by belugabehr** [comment]
Somewhat related (dovetails nicely) with my work on #1833

**Comment by nrg4878** [comment]
This change has been re-submitted as PR#1906 and merged. Closing this PR.

**Pull request closed without merging**